railties/lib/rails/generators/rails/benchmark/benchmark_generator.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rails/generators/named_base"
+
+module Rails
+  module Generators
+    class BenchmarkGenerator < NamedBase
+      IPS_GEM_NAME = "benchmark-ips"
+
+      argument :reports, type: :array, default: ["before", "after"]
+
+      def generate_layout
+        add_ips_to_gemfile unless ips_installed?
+        template("benchmark.rb.tt", "script/benchmarks/#{file_name}.rb")
+      end
+
+    private
+      def add_ips_to_gemfile
+        gem(IPS_GEM_NAME, group: [:development, :test])
+      end
+
+      def ips_installed?
+        in_root do
+          return File.read("Gemfile").match?(/gem.*\b#{IPS_GEM_NAME}\b.*/)
```

Comment:
Should we use `bundle show`?

Reply:
We could, I wonder which approach is faster? Probably the current one, but shelling out to `bundle show gem_name` is likely much less error prone.

Perhaps we could keep this approach now and refactor this in both the benchmark and db system generators.